### ISSUE TITLE
Enable shell command for pip

### ISSUE THREAD

**Issue by weiszr** [rationalmatter/Juno-Issue-Tracker#11]
Please enable shell commands and the ability to install packages via pip.

**Comment by navoshta** [comment]
Thanks @weiszr, are there any particular shell commands that you're missing? Adding all of them at once might be challenging, so it would be useful to know which ones to prioritise (apart from `pip`).

**Comment by weiszr** [comment]
Hello,

“pip” or “pip3” to install additional packages would be great. “ls” would be good to have as well.

cheers,

Robert


______________________________________________________________
Robert Weiss
Associate Professor of Natural Hazards
Department of Geosciences
Virginia Tech
4044 Derring Hall (0420)
Blacksburg, VA 24061, U.S.A.

Office: 1068A Derring
Phone:+1-540-231-2334
Fax: +1-540-231-3386

!!!!!Please note the new webpage!!!!!
http://coastalhazardsvt.weebly.com

Twitter: @VTCoastal

Albert Einstein (probably joking):
If the facts don't fit the theory, change the facts!

**Comment by holzschu** [comment]
You could use `ios_system` to include a large set of shell commands: https://github.com/holzschu/ios_system/

**Comment by navoshta** [comment]
Thanks @holzschu, I was going to take a closer look at it later.

**Comment by navoshta** [comment]
As a temporary substitute for `pip` command, Juno now comes with a [package installer](https://github.com/rationalmatter/Juno-Issue-Tracker/issues/37#issuecomment-529098898), which lets you install pure Python packages from PyPI.

I'll update the issue title to focus on the shell commands part.

**Comment by sjcorp** [comment]
I tried running pip freeze > requirements.txt and it gave me an error saying 'shell commands are restricted'. This is a fairly basic requirement, in order to deploy the notebook, say on Heroku. Can you please consider adding this feature. Thanks.

**Comment by navoshta** [comment]
Yep, thanks for reporting @sjcorp — sounds like it's more the case of missing `pip` rather than shell commands in general.

**Comment by Ishanthecoder** [comment]
I was trying to follow along with a lab and needed to install these (i just pasted the code below). Juno says "Shell commands are disabled in Juno" when i try to run it, any work around?

!pip install lenskit
!pip install -U numba

import lenskit.datasets as ds
import pandas as pd

!git clone https://github.com/chrash-course-ai/lab4-recommender-systems.git
data = ds.MovieLens('lab4-recommender-systems/')
print("Sucess! Installed sucessfully!")

**Comment by navoshta** [comment]
@Ishanthecoder Looks like the commands are trying to install lenskit and numba packages. Unfortunately, those either have native extensions, or depend on packages with native extensions, so you won't be able to install them due to OS restrictions, I'm afraid. To make those available in Juno, we may need to port them and embed in the app binary — please, submit package requests for the packages that you can't install, and we will get to them eventually.

**Comment by vphadke96** [comment]
I’m having trouble with installing pytorch. Any suggestions for the same?

**Comment by navoshta** [comment]
@vphadke96 You can't install pytorch I'm afraid, but there is a package request for it already: #35
